oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -187,15 +198,171 @@ def _find_hits(
         if consider_hits_from_input_region:
             # remove all hits where query and reference come from the same region
             search_results = search_results[
-                search_results["query_region_id"] != search_results["reference_region_id"]
+                search_results["query_region_id"]
+                != search_results["reference_region_id"]
             ]
 
         blast_table_hits = search_results.loc[
             search_results.alignment_length > search_results.min_alignment_length
         ]
-        oligos_with_hits = blast_table_hits["query"].unique()
 
-        return blast_table_hits, oligos_with_hits
+        return blast_table_hits
+
+    def get_references(
+        self, table_hits: pd.DataFrame, reference_database: ReferenceDatabase, region_id: str
+    ):
+        """
+        Retrieve the reference sequences from the search results.
+
+        :param table_hits: Dataframe containing the search results.
+        :type searchtable_hits_results: pd.DataFrame
+        :param reference_database: The reference database to compare against for specificity.
+        :type reference_database: ReferenceDatabase
+        :param region_id: The identifier for the region within the database to filter.
+        :type region_id: str
+        :return: Reference sequences
+        :rtype: list
+        """
+        # required_fields = [
+        #     "query",
+        #     "reference",
+        #     "alignment_length",
+        #     "query_start",
+        #     "query_end",
+        #     "query_length",
+        #     "query_sequence",
+        #     "reference_start",
+        #     "reference_end",
+        #     "reference_sequence",
+        #     "reference_strand"
+        # ]
+        # if not all(field in self.names_search_output for field in required_fields):
+        #     raise ValueError(
+        #         f"Some of the required fields {required_fields} are missing in the search results."
+        #     )
+        
+        # set the positions to a 0-based index
```

Comment:
Looks good overall, but I think for readability, it would be nice to have it into functions (like what you have in comments)